src/de.hpi.swa.graal.squeak/src/de/hpi/swa/graal/squeak/nodes/context/frame/FrameSlotWriteNode.java

```diff
@@ -66,6 +66,6 @@ protected final boolean isDoubleOrIllegal(final Frame frame) {
     }
 
     private static boolean verifyWrite(final Object value) {
-        return value != null && !(value instanceof Byte || value instanceof Integer || value instanceof Float);
```

Comment:
This was used to detect illegal writes of Java primitives not used in GraalSqueak to avoid that they propagate through the system. I'll turn it into an assertion.

Reply:
It's already an assertion, I hope I have clarified this better in the code with https://github.com/hpi-swa/graalsqueak/commit/3231e4b1c33418c1124e7451db87567b9cf78ecc.

Reply:
I am afraid this does not work for integer types, it only works for null and Boolean and presumably Float.
There are three tests that are failing because of this, and they are legitimate: for a method with an argument typed as Object, the argument comes as the value 40. Who knows where and who did the boxing, and how can the test for Byte and Short and Integer fail for it? Mind you, the value is also red from a slot, which also legitimately is the kind Object (everything on the stack is). So how does it gets boxed when read as on Object? Certainly not as a Long

Reply:
The tests are regressions from Squeak, and they are the PNGReadWriterTest failing ones

Reply:
Note that, at the very least, we also use the Int slot kind explicitly, for pc and stackp, and those values also legitimately get onto the stack

Reply:
Well, I saw it triggered by some testcases that worked after removing the check. But I confess I did not understand what was happening. And I did have a harrowing debugging experience when in my LargeInteger changes I returned a 0 instead of a 0L :)

Reply:
Yes, I'm still making this mistake sometimes. Integers are only used in very rare places and should never "escape" via the stack.